app/src/main/kotlin/io/homeassistant/companion/android/settings/gestures/views/GestureActionsView.kt

```diff
@@ -0,0 +1,96 @@
+import androidx.compose.foundation.selection.selectableGroup
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.RadioButton
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import io.homeassistant.companion.android.common.util.GestureAction
+import io.homeassistant.companion.android.settings.views.SettingsSubheader
+import io.homeassistant.companion.android.util.safeBottomPaddingValues
+
+/**
+ * View showing all actions for a gesture, grouped by category, and the
+ * currently configured action. Select to update the configured action.
+ *
+ * @param selectedAction The current action
+ * @param onActionSelected Called when an action is selected
+ */
+@Composable
+fun GestureActionsView(selectedAction: GestureAction, onActionSelected: (GestureAction) -> Unit) {
+    val actionsGrouped = GestureAction.entries.minus(GestureAction.NONE).groupBy { it.category }
+    LazyColumn(
+        contentPadding = safeBottomPaddingValues(applyHorizontal = false),
+        modifier = Modifier.selectableGroup(),
+    ) {
+        item {
+            // GestureAction.NONE is here so it's always first in the list with no header
+            GestureActionItem(
+                action = GestureAction.NONE,
+                checked = selectedAction == GestureAction.NONE,
+                onClick = { onActionSelected(GestureAction.NONE) },
+            )
+        }
+        actionsGrouped.forEach { (category, actions) ->
+            item {
+                SettingsSubheader(stringResource(category.description))
+            }
+            items(actions) { action ->
+                GestureActionItem(
+                    action = action,
+                    checked = selectedAction == action,
+                    onClick = { onActionSelected(action) },
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun GestureActionItem(action: GestureAction, checked: Boolean, onClick: () -> Unit) {
+    // Based on androidx.compose.material.samples.RadioGroupSample
+    Row(
+        Modifier
+            .fillMaxWidth()
+            .height(56.dp)
+            .selectable(
+                selected = checked,
+                onClick = onClick,
+                role = Role.RadioButton,
+            )
+            .padding(horizontal = 16.dp),
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        RadioButton(
+            selected = checked,
+            onClick = null, // Handled by parent Row
+        )
+        Text(
+            text = stringResource(action.description),
+            style = MaterialTheme.typography.body1.merge(),
+            modifier = Modifier.padding(start = 32.dp),
```

Comment:
Why 32dp here? Shouldn't we have the same padding on the left and right of the RadioButton?

Reply:
To align the text with the title in the toolbar, otherwise you have text at three different positions from the start which looks messy.